rust/src/lib.rs

```diff
@@ -182,3 +182,46 @@ extern "C" fn tokenizers_free(wrapper: *mut TokenizerWrapper) {
         drop(Box::from_raw(wrapper));
     }
 }
+
+#[no_mangle]
+extern "C" fn tokenizers_get_vocab_size(handle: *mut TokenizerWrapper, size: *mut usize) {
+    unsafe {
+        *size = (*handle).tokenizer.get_vocab_size(true);
+    }
+}
+
+#[no_mangle]
+extern "C" fn tokenizers_id_to_token(
+    handle: *mut TokenizerWrapper,
+    id: u32,
+    out_cstr: *mut *mut u8,
+    out_len: *mut usize,
+) {
+    unsafe {
+        let str = (*handle).tokenizer.id_to_token(id);
+        let c_str = match str {
+            Some(s) => CString::new(s).expect("Failed to create CString"),
+            None => CString::new("").expect("Failed to create CString"),
+        };
```

Comment:
Follow the same logic as decode,


store the data in internally allocated memory decode_str. This would avoid memory leak here (cstr get leaked).

The caller then to fetch the result using tokenizers_get_decode_str